audiobook/src/main/java/de/ph1b/audiobook/features/chapterReader/oggReading.kt

```diff
@@ -0,0 +1,165 @@
+package de.ph1b.audiobook.features.chapterReader
+
+import android.util.SparseArray
+import java.io.EOFException
+import java.io.InputStream
+import java.util.ArrayDeque
+
+private val OGG_PAGE_MAGIC = "OggS".toByteArray()
+
+internal data class OggPage(
+    val continuedPacket: Boolean,
+    val finishedPacket: Boolean,
+    val firstPageOfStream: Boolean,
+    val lastPageOfStream: Boolean,
+    val absoluteGranulePosition: Long,
+    val streamSerialNumber: Int,
+    val pageSequenceNumber: Long,
+    val packets: List<ByteArray>) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is OggPage) return false
+    return continuedPacket == other.continuedPacket &&
+        finishedPacket == other.finishedPacket &&
+        firstPageOfStream == other.firstPageOfStream &&
+        lastPageOfStream == other.lastPageOfStream &&
+        absoluteGranulePosition == other.absoluteGranulePosition &&
+        streamSerialNumber == other.streamSerialNumber &&
+        pageSequenceNumber == other.pageSequenceNumber &&
+        packets.size == other.packets.size &&
+        packets.indices.all { packets[it] contentEquals other.packets[it] }
+  }
+
+  override fun hashCode()
+      = listOf(continuedPacket.hashCode(), finishedPacket.hashCode(), firstPageOfStream.hashCode(),
```

Comment:
The argument is not weak, it's quite strong. Else an `OggPage` with 

`continuedPacket==true`, `finishedPacket==false` 

will have the same hash as 

`continuedPacket==false`, `finishedPacket==true`

but different equals.

So it's kind of a shortcut of the [Chapter hashcode implementation](https://github.com/PaulWoitaschek/MaterialAudiobookPlayer/blob/c1b7c7c168f934f93c32d8a3de33698c6a6d117c/audiobook/src/main/java/de/ph1b/audiobook/Chapter.kt#L37)